core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -451,10 +470,24 @@ public TableMetadata withUUID() {
       return this;
     } else {
       return new TableMetadata(null, formatVersion, UUID.randomUUID().toString(), location,
-          lastSequenceNumber, lastUpdatedMillis, lastColumnId, schema, defaultSpecId, specs, lastAssignedPartitionId,
-          defaultSortOrderId, sortOrders, properties, currentSnapshotId, snapshots, snapshotLog,
-          addPreviousFile(file, lastUpdatedMillis));
+          lastSequenceNumber, lastUpdatedMillis, lastColumnId, currentSchemaId, schemas, defaultSpecId, specs,
+          lastAssignedPartitionId, defaultSortOrderId, sortOrders, properties,
+          currentSnapshotId, snapshots, snapshotLog, addPreviousFile(file, lastUpdatedMillis));
+    }
+  }
+
+  private int reuseOrCreateNewSchemaId(Schema newSchema) {
+    // if the schema already exists, use its id; otherwise use the highest id + 1
+    int newSchemaId = currentSchemaId;
```

Comment:
Shouldn't this be `currentSchemaId + 1`?

Reply:
In the for loop below it will check if each existing schema's ID is greater or equal to this `newSchemaId` and +1 if so, so I think at the end of the method it will be currentSchemaId + 1 if no reuse happens?